### ISSUE TITLE
Block hard-coded credentials

### ISSUE THREAD

**Issue by dpauli** [kalessil/phpinspectionsea#495]
Its annoying to see code where someone makes this:

```php
$username = 'foo';
$password = 'bar';
```

Better reading passwords out of a credential manager which is connected to database or configuration file like:

```php
$username = Credentials::getEncryptedUser();
$password = Credentials::getEncryptedPassword();
```

I think it would be nice if PHP inspections mark variables like this (which get hardcoded value) as *not recommended*.

**Comment by funivan** [comment]
@dpauli how to detect if value is hardcoded ? 
`$defaultUser = 'test'`  Should we extract this value to the constant/method ?

**Comment by kalessil** [comment]
Probably this: https://github.com/dseguy/clearPHP/blob/master/rules/no-hardcoded-credential.md

**Comment by kalessil** [comment]
- ftp_connect
- ftp_login
- mysql_connect
- mysqli_connect
- ftp_login
- mssql_connect
- oci_connect
- imap_open
- cyrus_authenticate
- PDO::__construct
- ldap_bind

**Comment by kalessil** [comment]
//cc @dseguy: some new functions (starting from ldap_bind) for https://github.com/dseguy/clearPHP/blob/master/rules/no-hardcoded-credential.md

**Comment by dseguy** [comment]
Thanks for the ping. I added your suggestions, and found some more. 
I've made it a separate data-file, for future adding (I suspect I still left some somewhere...)
(Link to come soon).

**Comment by kalessil** [comment]
Oh, cool. Thank you, Damien.

**Comment by dseguy** [comment]
There it is : https://github.com/exakat/exakat/blob/master/data/php_logins.json
Feel free to add more :)

**Comment by kalessil** [comment]
Done
